src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java

```diff
@@ -33,9 +34,25 @@ public HearingManagementServiceImpl(final MessageSenderConfiguration messageSend
 
     @Override
     public void processRequest(String caseId, HearingDetailsRequest hearingDetailsRequest) {
-        cftHearingService.isValidCaseId(caseId);
+        Integer latestVersion = cftHearingService.getLatestVersion(caseId);
         isValidRequest(hearingDetailsRequest);
-        validateHmiHearingRequest(hearingDetailsRequest, caseId);
+        validateHmiHearingRequest(hearingDetailsRequest, caseId, latestVersion);
+    }
+
+    public boolean isValidRequestVersion(HearingDetailsRequest hearingDetailsRequest,
+                                         Integer latestHearingRequestVersion) {
+        Integer hearingCaseVersionId = 0;
```

Comment:
The `CftHearingServiceImpl.getLatestVersion()` method returns `0` if it can't find the header. This method could pass if that occurred and the case version ID wasn't present, by the looks of this.

We should probably be throwing an error earlier on in `getLatestVersion()` if we can't find the header in the first place, rather than defaulting `0` there. That would make this a bit more reliable.

Reply:
Agreed